nautobot/core/graphql/types.py

```diff
@@ -34,3 +37,20 @@ class ContentTypeType(OptimizedNautobotObjectType):
 
     class Meta:
         model = ContentType
+
+
+class DateType(graphene.Date):
+    """
+    Overriding the default serialize method from https://github.com/graphql-python/graphene/blob/master/graphene/types/datetime.py
+    to handle the case where the date object is passed as a str object.
+    """
+
+    @staticmethod
+    def serialize(date):
+        if isinstance(date, (datetime.date, datetime.datetime)):
+            date = date.date()
```

Comment:
This won't work on a `datetime.date` object:

```py
>>> d = datetime.date.today()
>>> d
datetime.date(2024, 2, 1)
>>> d.date()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'datetime.date' object has no attribute 'date'
```

Reply:
Should be fixed in the latest commit, sorry for the trouble.

Reply:
Looks good. We should probably add a test just for this, providing string, date and datetime instances for the serialize method.